script/modelling/space_shuttle_reentry_trajectory.jl

```diff
@@ -0,0 +1,281 @@
+#'          R_e & = 20902900,                                    \qquad & c_2 & =  0.21286289 \times 10^{-3}, \\
+#'            S & = 2690,                                        \qquad & c_3 & = -0.10117249 \times 10^{-5}.
+#' \end{align}
+#' $$
+
+#' The reentry trajectory begins at an altitude where the aerodynamic forces are quite small
+#' with a weight of $w = 203000$ (lb) and mass $m = w / g_0$ (slug), where $g_0 = 32.174$ (ft/sec$^2$).
+#' The initial conditions are as follows:
+#' 
+#' $$
+#' \begin{align}
+#'      h & = 260000 \text{ ft},  \qquad & v      & = 25600 \text{ ft/sec}, \\
+#'   \phi & = 0      \text{ deg}, \qquad & \gamma & = -1    \text{ deg},    \\
+#' \theta & = 0      \text{ deg}, \qquad & \psi   & = 90    \text{ deg}.
+#' \end{align}
+#' $$
+
+#' The final point on the reentry trajectory occurs at the unknown (free) time $t_F$,
+#' at the so-called terminal area energy management (TAEM) interface, which is defined by the conditions
+#' 
+#' $$
+#' h = 80000 \text{ ft}, \qquad v = 2500 \text{ ft/sec}, \qquad \gamma = -5 \text{ deg}.
+#' $$
+
+#' As explained in the book, our goal is to maximize the final crossrange,
+#' which is equivalent to maximizing the final latitude of the vehicle,
+#' i.e., $J = \theta(t_F)$.
+
+#' ![Max crossrange shuttle reentry](https://i.imgur.com/1tUN3wM.png)
+
+#' ### Approach 
+#' We will use a discretized model of time, with a fixed number of discretized points, $n$.
+#' The decision variables at each point are going to be the state of the vehicle and the controls commanded to it.
+#' In addition, we will also make each time step size $\Delta t$ a decision variable;
+#' that way, we can either fix the time step size easily,
+#' or allow the solver to fine-tune the duration between each adjacent pair of points.
+#' Finally, in order to approximate the derivatives of the problem dynamics, we will use either rectangular
+#' or [trapezoidal](http://en.wikipedia.org/wiki/Trapezoidal_rule) integration.
+
+#' ### Disclaimer
+#' Do not try to actually land a Space Shuttle using this notebook! 😛
+#' There's no mesh refinement going on, which can lead to unrealistic trajectories having position
+#' and velocity errors with orders of magnitude $10^4$ ft and $10^2$ ft/sec, respectively.
+
+using Interpolations, Plots
+using JuMP, Ipopt  # , KNITRO
+
+#+
+
+# Global variables
+const w  = 203000  # weight (lb)
+const g₀ = 32.174  # acceleration (ft/sec^2)
+const m  = w / g₀  # mass (slug)
+
+# Aerodynamic and atmospheric forces on the vehicle
+const ρ₀ =  0.002378
+const hᵣ =  23800.0
+const Rₑ =  20902900.0
+const μ  =  0.14076539e17
+const S  =  2690.0
+const a₀ = -0.20704
+const a₁ =  0.029244
+const b₀ =  0.07854
+const b₁ = -0.61592e-2
+const b₂ =  0.621408e-3
+const c₀ =  1.0672181
+const c₁ = -0.19213774e-1
+const c₂ =  0.21286289e-3
+const c₃ = -0.10117249e-5
+
+#+
+
+user_options_ipopt = (
+    "mu_strategy" => "monotone",
+    "linear_solver" => "ma27",
```

Comment:
Same comment as below about ma27. I am wondering whether we should include a comment there